specification/openai/resource-manager/Microsoft.OpenAI/preview/2021-09-01-preview/openai.json

```diff
@@ -0,0 +1,1990 @@
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Returns with a response body containing the subscription keys for the resource: key1\nThe first API key for authentication when client calls the API endpoint.\n\nkey2\nThe second API key for authentication when client calls the API endpoint.",
+            "schema": {
+              "$ref": "#/definitions/ApiKeys"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.OpenAI/accounts/{accountName}/regenerateKey": {
+      "post": {
+        "tags": [
+          "OpenAIAccounts"
+        ],
+        "operationId": "Accounts_RegenerateKey",
+        "description": "Regenerates the specified account key for the specified OpenAI account.",
+        "x-ms-examples": {
+          "Regenerate Keys": {
+            "$ref": "./examples/RegenerateKey.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/accountNameParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "x-ms-client-flatten": true,
+            "schema": {
+              "$ref": "#/definitions/RegenerateKeyParameters"
+            },
+            "description": "regenerate key parameters."
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "",
+            "schema": {
+              "$ref": "#/definitions/ApiKeys"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.OpenAI/skus": {
```

Comment:
>skus [](http://example.com/codeflow?start=64&length=4)

Is it specific for subscription or location?
Normal pattern is to model it as a location based API rather than defining checkSkuAvailability

Reply:
Chirag, we have actually leverage ARM SKU API, so this API actually is from ARM: https://microsoft.sharepoint.com/:w:/r/teams/azureresourcemanagerteam/_layouts/15/Doc.aspx?sourcedoc=%7BD68FD846-9EC2-4D20-925D-133626DD6D1B%7D&file=arm-sku-api.docx&action=default&mobileredirect=true&share=IQFG2I_Wwp4gTZJdEzYm3W0bATGtnXbTsaqMOdZ4ErrL-Jo